plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XProjectManager.java

```diff
@@ -198,11 +195,13 @@ protected XBuildResult doBuild(Set<URI> dirtyFiles, Set<URI> deletedFiles,
 		XBuildResult result = incrementalBuilder.build(request);
 
 		projectStateHolder.updateProjectState(request, result);
+		if (doGenerate && !result.getAffectedResources().isEmpty()) {
```

Comment:
Two questions about this:
- the build request does know if doGenerate is true or false. The build result knows the affected resources. Should the projectStateHolder decide on its own, if it should persist? E.g. should this logic be moved into the updateProjectState call above?
- projectStateHolder.writeProjectState has an internal guard that it should not write if the hashed contents is empty. Why is that?

Reply:
Regarding #1: I changed that, good idea. However, it was necessary to change the XBuildRequest also.
Regarding #2: I do not remember the reason. I removed it and test if it still works.